### PR TITLE
Fixed some clippy warnings in components

### PR DESCRIPTION
**Pull request by komutumu**
This PR fixes some clippy warnings by; 

- Removing unused lifetimes
- Simplifying boolean expressions
- Removing useless use of format!
- Replacing the use Option.and_then() with .map()
- Replacing the call of Option.map().flatten() with .and_then()

---
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes are part of #31500 
- [X] These changes do not require tests because it doesn't touch functionality.

### CODE REVIEW

**Comment by jahielkomu** [comment]
Hello @mrobinson 
Please kindly review.
Thanks.

**Comment by jahielkomu** [comment]
@mrobinson , a lot of the warnings talk about the use of unsafe code, should we use ```#[allow(unsafe_code)]``` ? 
Please advise.

File: components/script/dom/range.rs [hunk 1]
```diff
@@ -1189,9 +1189,8 @@ impl WeakRangeVec {
                 let move_start = node_is_start && range.start_offset() == offset;
                 let move_end = node_is_end && range.end_offset() == offset;
 
-                let remove_from_node = move_start && move_end ||
-                    move_start && !node_is_end ||
-                    move_end && !node_is_start;
+                let remove_from_node =
+                    !(!move_start || !move_end && node_is_end) || move_end && !node_is_start;
```
Comment (mrobinson) [inline components/script/dom/range.rs:1193]: Hrm...I think this new formulation is more difficult to understand. In this case, I would add an exception for the clippy warning or rewrite the logic to be more understandable.

File: components/script/dom/range.rs [hunk 2]
```diff
@@ -1252,9 +1251,8 @@ impl WeakRangeVec {
                 let move_start = node_is_start && start_offset > offset;
                 let move_end = node_is_end && end_offset > offset;
 
-                let remove_from_node = move_start && move_end ||
-                    move_start && !node_is_end ||
-                    move_end && !node_is_start;
+                let remove_from_node =
+                    !(!move_start || !move_end && node_is_end) || move_end && !node_is_start;
```
Comment (mrobinson) [inline components/script/dom/range.rs:1255]: I feel the same way about this new code.

**Review by mrobinson — CHANGES_REQUESTED** [review]

File: components/script/dom/range.rs (hunk 1 above)
Comment (jahielkomu) [reply components/script/dom/range.rs:1193]: I absolutely agree with you @mrobinson ...I picked this straight from the clippy suggestion and felt it was more complicated but thought it was the standard. I checked that it evaluated to the same as the before and applied it.
Let me find another way for this sir.
Thank you.

File: components/script/dom/range.rs (hunk 2 above)
Comment (jahielkomu) [reply components/script/dom/range.rs:1255]: Let me work on it sir

**Comment by jahielkomu** [comment]
I have updated the PR @mrobinson , kindly review again.

**Review by mrobinson — APPROVED** [review]

**Comment by jahielkomu** [comment]
Hmm something seems to not be going fine sir @mrobinson .
 Is it due to the changes made in the PR?
```./mach test-unit```, ```./mach test-tidy``` and ```./mach build -d``` ran with no errors

**Comment by jahielkomu** [comment]
> @mrobinson , a lot of the warnings talk about the use of unsafe code, should we use `#[allow(unsafe_code)]` ? Kindly advise.

Sir would you recommend this?

**Comment by mrobinson** [comment]
> > @mrobinson , a lot of the warnings talk about the use of unsafe code, should we use `#[allow(unsafe_code)]` ? Kindly advise.


Adding `allow(unsafe_code)` shouldn't be necessary, as it's absence would cause a compilation error. I think it would be helpful to see the actual error messages that you are referring to in the clippy warnings. If they are the ones regarding the missing "# Safety" section in the documentation, the fix is to write that documentation. That requires a bit of understanding and care though, as we want the documentation to actually reflect the real safety requirements of those functions.

**Pull request merged**